apps/emqx_utils/src/emqx_variform_bif.erl

```diff
@@ -47,15 +45,37 @@
     replace/4,
     regex_match/2,
     regex_replace/3,
+    regex_extract/2,
     ascii/1,
     find/2,
     find/3,
     join_to_string/1,
     join_to_string/2,
     unescape/1,
-    nth/2
+    any_to_str/1
 ]).
 
+%% Array functions
+-export([nth/2]).
+
+%% Control functions
+-export([coalesce/1, coalesce/2]).
```

Comment:
Wondering: don't we want to make `coalesce` lazy?

Also, a question: how is a customer intended to emulate `IFF(.., .., ...)` (functional if)? Via `coalesce`?

Reply:
> Wondering: don't we want to make coalesce lazy?
Good idea, maybe for improvement. The `eval` loop can be enhanced.

> emulate IFF(cond, a, b)

It's easy: `coalesce([coalesce([a, '']), b])`

Reply:
can add a `IFF` function

Reply:
To make it more complete, we can:

- `coalesce` should be a reserved word in the expression token scanner
- will need to add `iif` function (reserved word too) for if this else that
- to support `iif`, we'll need more compare functions, such as `gt`, `gte`, `eq`, `lt`, `lte`.
- then there is perhaps to be match functions too

Then if we add variable bindings, it'll become even more complete.

Finally allow user defined functions to be a programing language.

Reply:
The idea of the use case that occurred to me while reviewing was checking on emptiness in the `iif` condition.